### ISSUE TITLE
Thumbnail generation low res

### ISSUE THREAD

**Issue by Xphine** [algernon-A/Ploppable-RICO-Revisited#43]
First of all, I did notice an issue during the switch from old RICO. I have unsubscribed to the old one, subscribed to this before starting the game. The first load of the save file, RICO was not running properly with all places RICO build all gone or reverted back to unique buildings. The mod interface was blank with no building in it. I thought I didn't enable Harmony properly but that's not the case since I have a number of mods that also utilise it. After the second load, the mod seems to work fine but something might already go bonker.

![000](https://user-images.githubusercontent.com/49283095/82146585-8cdb4f80-9875-11ea-89fa-ee95f580e3a6.jpg)
This one has Nanachi citizen protruding from it.
Workshop link: https://steamcommunity.com/sharedfiles/filedetails/?id=1200076366&searchtext=

![001](https://user-images.githubusercontent.com/49283095/82146587-906ed680-9875-11ea-8971-d7abdf162c77.jpg)

Also to note that this mod auto-generated will not work with this particular asset.
https://steamcommunity.com/sharedfiles/filedetails/?id=1791879291
Since all of the lod and stuff are underground.

Is there a way to delete those thumbnails cache and force the mod to remake it?

**Comment by algernon-A** [comment]
The Nanachi citizen issue is with Find It's panel, not this one - it's a known issue with props and Find It, and has to do with the way Find It generates and displays its images.  I don't know enough about Find It to be able to help with that one, sorry.

For this mod, currently thumbnails are auto-generated on asset load; with 2.1 I'm switching that to generation after all loading is complete.  I'm hoping that will resolve any race condition issues with rendering that might be going on (it also will enable the option of regeneration).

Which mod are you using for the expanded panel?  Does this problem still happen without it (send a screenshot)?

**Comment by Xphine** [comment]
I'm using Resize it!
https://steamcommunity.com/sharedfiles/filedetails/?id=1577882296&searchtext=resize+it

No, after disabling it nothing changes. I also tried to disable some more but all seem to be unrelated.
![image](https://user-images.githubusercontent.com/49283095/82172110-78d63300-98f3-11ea-8e9d-e08066b0f1c3.png)

I'm going to check Harmony dependency mods next.

**Comment by algernon-A** [comment]
Just a thought - are you using any mods that might affect rendering resolution or aliasing, e.g. Dynamic Resolution, Render It!, PostProcessFX?

**Comment by Xphine** [comment]
Would you believe it? That low resolution is actually related to the in-game Texture Quality option. None of the mods have anything to do with it. And that huge red dot is actually the result of this map theme.

https://steamcommunity.com/sharedfiles/filedetails/?id=1760851468&

The red color around the edge of the buildings seems to be related to the theme lighting somehow. Don't tell me that dot was a light source lol.

**Comment by Sipke82** [comment]
i guess this might be related:
![image](https://user-images.githubusercontent.com/30873812/82928628-bc2f4200-9f82-11ea-8f20-cb445d48eb2f.png)

**Comment by Xphine** [comment]
@Sipke82 That's odd, sky background, did you use some skyscape mod?

**Comment by Sipke82** [comment]
I use the following mods:
https://steamcommunity.com/sharedfiles/filedetails/?id=1565457981

Interestingly, if i use this mod list on my other Steam account, im getting regular icons. Perhaps it depends on some kind of loading order.....

**Comment by Xphine** [comment]
On a different Steam account... you mean loading the same save? Your mod list doesn't look too different from mine much. Although someone doesn't recommend Instant exit to desktop mod as some other mods require downtime when you exit to write some cache.

Also, if it's a different map on the same mod list, I suggest you change the map theme for testing. And any mod related to lighting might also be causing the problems (not on its own but when parameters were changed).

**Comment by Sipke82** [comment]
Main
---
- Mods: https://steamcommunity.com/sharedfiles/filedetails/?id=1565457981
- Assets: 3000
- Savegame of a city
- borked icons

Other account
---
- Mods: https://steamcommunity.com/sharedfiles/filedetails/?id=1565457981
- Assets: 0
- New game
- OK icons

**Comment by algernon-A** [comment]
I've just pushed version 2.1, which (amongst other things) fixes the SoCal Laguna Homes rendering issue.  It also incorporates a number of thumbnail rendering changes that may or may not help with the 'white asset' problem - please let me know if you're still getting that after the update.

Thanks for the mod listings - I'll do another round of investigation.

**Comment by Xphine** [comment]
![image](https://user-images.githubusercontent.com/49283095/83094838-9746e600-a0cc-11ea-954f-6346b5018108.png)
A lot less red, thanks, but the semicircles still remain. Don't know what wrong with that Damac  https://steamcommunity.com/sharedfiles/filedetails/?id=1627657698&

![image](https://user-images.githubusercontent.com/49283095/83095106-d4ab7380-a0cc-11ea-8133-b486b3b3fa90.png)
In Medium Texture Quality setting the thumb is awfully in low res. You should probably warn people about it if it can't be easily fixed soon.

**Comment by algernon-A** [comment]
I'll see what more I can do about the textures.  Those seimcircles are really odd - what map theme are you using?

**Comment by Xphine** [comment]
The one I mentioned before.
https://steamcommunity.com/sharedfiles/filedetails/?id=1760851468&

**Comment by algernon-A** [comment]
Can you also post your mods list, and any settings of any graphics mods you might be using (e.g. relight, ultimate eyecandy, etc.)?

**Comment by Sipke82** [comment]
White icons fixed for me  with the new update!

**Comment by Xphine** [comment]
My mod list: https://steamcommunity.com/sharedfiles/filedetails/?id=2112400477

I don't use Theme mixer and Relight. My Ultimate Eye Candy setting is just normal, well you can't adjust anything much with it anyway without Relight. I'm using Satellite LUT by the way If you are curious.
![image](https://user-images.githubusercontent.com/49283095/83252376-51c21000-a1d5-11ea-9a8c-ac91568fad73.png)

It's all came down to the theme. I use this mod to switch theme: https://steamcommunity.com/sharedfiles/filedetails/?id=611254368
And it's what makes me realize that it's really the theme that has a problem during auto-generate.

**Comment by algernon-A** [comment]
At this point, I consider these issues addressed with the various configuration options (and rendering updates) introduced since the original issue was raised.  Closing; please submit a new issue if problems persist.
